plugins/lookup/passwordstore.py

```diff
@@ -261,6 +260,13 @@ def check_pass(self):
                     if ':' in line:
                         name, value = line.split(':', 1)
                         self.passdict[name.strip()] = value.strip()
+
+            passfile = os.path.join(self.paramvals['directory'], self.passname + ".gpg")
+            if not os.path.isfile(passfile):
+                # A password was returned, but there is no .gpg file for it, so
+                # this is probably a tree node or something undefined is going on.
+                raise AnsibleError('Passwordstore returned password, but \'{0}\' does not exist'.format(passfile))
```

Comment:
@felixfontein I don’t think the logic in the if block applies (as it’s not the same as "missing"). That’s why I implemented the exception - this really should be exceptional, happening by accident. As a user of this role, I won’t assume "missing" == there was a subtree path of the same name.

Reply:
So it is not possible to have a secret of the same name as a directory? Since the secret is stored in `secretname.gpg`, I would assume you can have both. (But then I never used passwordstore, so this could be completely wrong.)

Reply:
Hm, you're making a really good point here that didn't occur to me. It also means that `pass` really has a problem here (it's not possible to list the subtree of xyz if there is also a gpg file xyz.gpg).

Given that, I think you're right, the same logic could be applied, Chapeau! So let me see how to change this so it works as expected (still feels really ugly though, I wished that `pass ls` and `pass show` would be different operations like `ls` and `cat`). Anyway...